### ISSUE TITLE
feat: support pretokenised datasets

### ISSUE THREAD

**Issue by kmehant** [foundation-model-stack/fms-hf-tuning#191]
### Interface changes 
Introduction of an additional flag `--pretokenized` which can default to `false`

### Motivation
We should be supporting datasets that are pretokenized as input to fms-hf-tuning which involve
1. modifying the tokenizer like adding special tokens outside the scope of the fms-hf-tuning
2. pretokenizing the dataset so the datasets that are given as input to the fms-hf-tuning repo would hold `input_ids`, `labels`, and optionally `attention_mask`

### Changes needed

1. We should skip fms-hf-tuning specific code on adding special tokens to tokenizer etc. when the flag `--pretokenized` is passed.
2. Should allow for providing data in tokenized format - having input_ids and labels. 
3. Should choose appropriate data collator and also should force providing custom tokenizer path.

**Comment by ashokponkumar** [comment]
@alex-jw-brooks @Ssukriti @kmehant 

Can we possibly work together and see what is the best way to achieve this? 

Also I see a couple of PRs which are already in place to address this. Also, trl sft trainer seems to have added support for [this](https://github.com/huggingface/trl/pull/1520).
